contentful/apps
# feat: Add ContentType and preview path selection on config page

## Purpose

This PR adds the beginnings of the ContentType/PreviewPath selection section on the config page. 

Designs: https://www.figma.com/file/3zhaUVyNKPZOFjVPL7mG00/Vercel?type=design&node-id=758-42795&mode=design&t=ipC1WnpgyJcr4PdA-4

## Approach

I created separate components for the `contentType` and `previewPath` row of selection and the associated list, which is rendered from the stored reducer state. 

https://github.com/contentful/apps/assets/58186851/609cc2e3-86af-45b5-9169-a993ea65c7e1

## Testing steps

Render the Vercel (development) config page using this branch locally. Go ahead and add a contentType and previewPath configuration, play around with removing rows and adding them, and save the configuration. The rows should then persist if you refresh the page. 

## Follow-up work

The following are TO DO items to improve on as we build on top of the steel thread, after consulting with Emily & team: 
1. Flesh out the workflow around when to enable the Add Content Type button. Currently, it is disabled if the current row is not filled out, and the and is disabled if there exists an added row already. This prevents the user from adding a bunch of unnecessary rows if they have not filled out the ones already added, but am open to making tweaks to this flow! 
2. Add logic to either to potentially prevent configuring multiple `previewPaths` per `contentType`.
